src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseDefaultDllImportSearchPathsAttributeTests.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    // All the test cases use user32.dll as an example,
+    // however it is a commonly used system dll and will be influenced by Known Dlls mechanism,
+    // which will ignore all the configuration about the search algorithm.
+    // Fow now, this rule didn't take Known Dlls into consideration.
+    // If it is needed in the future, we can recover this rule.
+    public class UseDefaultDllImportSearchPathsAttributeTests : DiagnosticAnalyzerTestBase
+    {
+        // It will try to retrieve the MessageBox from user32.dll, which will be searched in a default order:
+        //  1. The directory from which the application is loaded.
+        //  2. The current directory.
+        //  3. The system directory, usually C:\\Windows\\System32\\ (The GetSystemDirectory function is called to obtain this directory.).
+        //  4. The 16-bit system directory – There is no dedicated function to retrieve the path of this directory, but it is searched as well.
+        //  5. The Windows directory. The GetWindowsDirectory function is called to obtain this directory.
+        //  6. The directories that are listed in the PATH environment variable
+        [Fact]
+        public void Test_DllImportAttribute_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Runtime.InteropServices;
+
+class TestClass
+{
+    [DllImport(""user32.dll"")]
+    public static extern int MessageBox(IntPtr hWnd, String text, String caption, uint type);
+
+    public void TestMethod()
+    {
+        MessageBox(new IntPtr(0), ""Hello World!"", ""Hello Dialog"", 0);
+    }
+}",
+            GetCSharpResultAt(8, 30, UseDefaultDllImportSearchPathsAttribute.Rule, "MessageBox"));
+        }
+
+        // [DllImport] is set with an absolute path, which will let the [DefaultDllImportSearchPaths] be ignored.
+        // So user32.dll will also be searched in the default order.
+        [Fact]
```

Comment:
It doesn't just load from the absolute path?

Reply:
You are right, wrong comment.